scala-package/core/src/main/scala/org/apache/mxnet/javaapi/NDArray.scala

```diff
@@ -172,6 +189,18 @@ object NDArray extends NDArrayBase {
   = org.apache.mxnet.NDArray.array(
     sourceArr.asScala.map(ele => Float.unbox(ele)).toArray, shape, ctx)
 
+  /**
+    * Create a new NDArray that copies content from source_array.
+    * @param sourceArr Source data (list of Doubles) to create NDArray from.
+    * @param shape shape of the NDArray
+    * @param ctx The context of the NDArray, default to current default context.
+    * @return The created NDArray.
+    */
+  def arrayWithDouble(sourceArr: java.util.List[java.lang.Double], shape: Shape,
```

Comment:
Does anything prevent us from using a java.util.Collection instead of a list? If we can do that it'll also let people use the same method with Sets, queues, etc.

Reply:
That seems a good suggestion. Basically we should use Iterable as a super class for this method, rather than a subclass. 

I can take it up in a separate PR other than this. 
I created a backlog JIRA item to track this : https://issues.apache.org/jira/browse/MXNET-1293